src/store/actions.js

```diff
@@ -0,0 +1,12 @@
+import { zomatoApi } from "./api.js";
+
+export function getData({ commit }) {
+  commit("setLoadingState", "Loading");
+  zomatoApi
+    .get()
+    .then(response => commit("setRestaurantsData", response))
+    .catch(error => {
+      commit("setLoadingState", "Error");
+      console.error(error);
+    });
```

Comment:
I highly recommend you using `async` instead of then-catch. Here, if `then` throws an exception, it's still handled by `catch`, despite the fact it's a totally different source of error --- as you know, considering your implementation of `"setRestaurantsData"`.